libr/io/undo.c

```diff
@@ -83,9 +83,9 @@ R_API void r_io_sundo_push(RIO *io, ut64 off, int cursor) {
 	if (!io->undo.s_enable) {
 		return;
 	}
-	// the first insert
-	if (io->undo.idx > 0) {
-		undo = &io->undo.seek[io->undo.idx - 1];
+	// don't push duplicate seek
+	if (io->undo.undos > 0) {
+		undo = &io->undo.seek[(io->undo.idx - 1 + R_IO_UNDOS) % R_IO_UNDOS];
```

Comment:
not sure if we want to make the undo history cyclic

Reply:
Are we thinking about the same thing? It was already storing history in a cyclic buffer before my changes but with bugs. There were even tests expecting this.  With fixed size history deleting the oldest entry  when there is no space for new ones is only sane behaviour.